lifetiles-graph/src/main/java/nl/tudelft/lifetiles/annotation/controller/BookmarkController.java

```diff
@@ -0,0 +1,181 @@
+
+    /**
+     * the searchbox to search through all annotations.
+     */
+    @FXML
+    private TextField searchBox;
+
+    /**
+     * The list of all bookmarks.
+     */
+    private ObservableList<AbstractBookmark> listItems;
+    /**
+     * The filtered list that will wrap the listItems.
+     * do not add items directly to this list, instead add them to listItems.
+     */
+    private FilteredList<AbstractBookmark> filteredItems;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public final void initialize(final URL location,
+            final ResourceBundle resources) {
+        hide();
+
+        // wrap the observable list in a filtered list so we can search it later
+        listItems = FXCollections.observableArrayList();
+        filteredItems = new FilteredList<AbstractBookmark>(listItems);
+        bookmarkList.setItems(filteredItems);
+
+        /*
+         * create a new Cell factory so we can use a different method
+         * to create the text of each cell,
+         * and assign an event handler for mouse clicks.
+         */
+        bookmarkList.setCellFactory(new Callback<ListView<AbstractBookmark>,
+                ListCell<AbstractBookmark>>() {
+            /**
+             * {@InheritDoc}
+             */
+            @Override
+            public ListCell<AbstractBookmark> call(
+                    final ListView<AbstractBookmark> param) {
+                final ListCell<AbstractBookmark> cell = new ListCell<AbstractBookmark>() {
+                    /**
+                     * the ListCell, creates the correct text and adds the right eventhandler.
+                     */
+                    public void updateItem(final AbstractBookmark item,
+                            final boolean empty) {
+                        super.updateItem(item, empty);
+                        if (item != null) {
+                            setText(item.toCellString());
+
+                            setOnMouseClicked(event -> {
+                                if (event.getClickCount() == 2) {
+                                    shout(Message.GOTO, "" ,
+                                            Long.valueOf(item.getUnifiedPosition()));
+                                }
+                            });
+                        }
+                    }
+                };
+                return cell;
+            }
+        });
+        initListeners();
+    }
+
+
+    /**
+     * Initialize the shout listeners.
+     */
+    // java doesn't let us typecheck generics so the casts are unchecked
+    // checkstyle doesn't like assert without parentheses with generics
+    @SuppressWarnings({"checkstyle:genericwhitespace", "unchecked"})
+    private void initListeners() {
+        listen(Message.BOOKMARKS, (sender, subject, args) -> {
+            show();
+        });
+
+        listen(Message.LOADED, (sender, subject, args) -> {
+
+            switch (subject) {
```

Comment:
Redundant code in the cases, you should only set the list item to the appropriate value inside the case body.